### PR TITLE
feat(content-docs): version path & prefix configuration options

### PR DESCRIPTION
**Pull request by Jersyfi**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

My personal motivation for modifying the `docusaurus-plugin-content-docs` plugin is to have more flexibility in the file structure. With this feature it is possible to group all files in one folder without having them in the project root folder.

## Test Plan

I have added a test folder called `versioned-site-with-custom-path` with the `versionPath` and `versionPrefix` options set and all files grouped together. In this state the tests are simple and I need your opinion if any changes are needed.

image

### Test links

Deploy preview: https://deploy-preview-8737--docusaurus-2.netlify.app/

## Options

* `versionPath`
type: string | undefined
default: undefined
* `versionPrefix`
type: boolean
default: true

## Mofified Documentation

Path: website/docs/api/plugins/plugin-content-docs.mdx
Deploy preview: https://deploy-preview-8737--docusaurus-2.netlify.app/docs/api/plugins/@docusaurus/plugin-content-docs/

## Related issues/PRs

https://github.com/facebook/docusaurus/issues/8061

### CODE REVIEW

File: website/docs/api/plugins/plugin-content-docs.mdx [hunk 1]
```diff
@@ -34,6 +34,8 @@ Accepted fields:
 | Name | Type | Default | Description |
 | --- | --- | --- | --- |
 | `path` | `string` | `'docs'` | Path to the docs content directory on the file system, relative to site directory. |
+| `versionPath` | `string` |  `''` | Path to versioned files on filesystem, relative to site dir. |
```
Comment (Jersyfi) [inline website/docs/api/plugins/plugin-content-docs.mdx:37]: ## TODO
Found that i missed changing the `versionPath` default value from `''` to `undefined`.
image

File: packages/docusaurus-plugin-content-docs/src/versions/files.ts [hunk 2]
```diff
@@ -21,35 +21,53 @@ import type {
...
+  pluginId: string,
+  versionPrefix: boolean,
+): string {
+  return pluginId === DEFAULT_PLUGIN_ID || versionPrefix === false
     ? fileOrDir
     : `${pluginId}_${fileOrDir}`;
 }
 
-/** `[siteDir]/community_versioned_docs/version-1.0.0` */
+/** `[versionPath]/[pluginId?]_versioned_docs/version-1.0.0` */
```
Comment (Jersyfi) [inline packages/docusaurus-plugin-content-docs/src/versions/files.ts:45]: Change the following comment word.
versionPath -> versionSiteDir

File: packages/docusaurus-plugin-content-docs/src/versions/files.ts [hunk 3]
```diff
@@ -21,35 +21,53 @@ import type {
...
-    siteDir,
-    addPluginIdPrefix(VERSIONED_DOCS_DIR, pluginId),
+    addVersionDir(siteDir, versionPath),
+    addPluginIdPrefix(VERSIONED_DOCS_DIR, pluginId, versionPrefix),
     `version-${versionName}`,
   );
 }
 
-/** `[siteDir]/community_versioned_sidebars/version-1.0.0-sidebars.json` */
+/** `[versionPath]/[pluginId?]_versioned_sidebars/version-1.0.0-sidebars.json` */
```
Comment (Jersyfi) [inline packages/docusaurus-plugin-content-docs/src/versions/files.ts:60]: Change the following comment word.
versionPath -> versionSiteDir

File: packages/docusaurus-plugin-content-docs/src/versions/files.ts [hunk 4]
```diff
@@ -75,9 +93,17 @@ export function getDocsDirPathLocalized({
   });
 }
 
-/** `community` => `[siteDir]/community_versions.json` */
-export function getVersionsFilePath(siteDir: string, pluginId: string): string {
-  return path.join(siteDir, addPluginIdPrefix(VERSIONS_JSON_FILE, pluginId));
+/** `community` => `[versionPath]/community_versions.json` */
```
Comment (Jersyfi) [inline packages/docusaurus-plugin-content-docs/src/versions/files.ts:96]: Change the following comment word.
versionPath -> versionSiteDir

**Review by slorber — CHANGES_REQUESTED** [review]
Looks like a good first draft

That probably solves your use-case, but I think this API is a bit awkward and not the most flexible either.

Reading the docs, it's not so clear to me what `versionPrefix` even is 😅

I'd prefer if we worked on designing the API we want first before doing the implementation.

For now, I think this API is more flexible than yours and less confusing:

```js
[
  "@docusaurus/plugin-content-docs",
  {
    id: "ios",
    path: "./ios-docs/v-next",
    versionPath: ({version}) => `./ios-docs/v-${version.toLowerCase().replace(".","-"}`,
  },
];
```

**Comment by Jersyfi** [comment]
Closing the PR as there is no time for further changes.

**Pull request closed without merging**